### PR TITLE
chore: update vyper imports

### PR DESCRIPTION
**Pull request by tserg**
Update imports from Vyper as a result of https://github.com/vyperlang/vyper/pull/2974, https://github.com/vyperlang/vyper/pull/3182, https://github.com/vyperlang/vyper/pull/3209

Also fixes test path and updated README.

### CODE REVIEW

**Review by charles-cooper — COMMENTED** [review]
looks good but holding off on merging this as a lot of other changes to the semantics pass are coming up for vyper

File: boa/vyper/contract.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ def _dealias(self, maybe_address):
             return maybe_address
 
     def get(self, truncate_limit=None):
-        if isinstance(self.typ, MappingType):
+        if is_type_t(self.typ, HashMapT):
```
Comment (charles-cooper) [inline boa/vyper/contract.py:366]: these are not right. `is_type_t` checks if the argument is a TYPE_T with the second argument as its typedef. this should probably be `if isinstance(self.typ, HashMapT)`

File: boa/vyper/contract.py [hunk 2]
```diff
@@ -386,7 +386,7 @@ def get(self, truncate_limit=None):
                     # decode aliases as needed/possible
                     dealiased_path = []
                     for p, t in zip(path, path_t):
-                        if is_base_type(t, "address"):
+                        if is_type_t(t, AddressT):
```
Comment (charles-cooper) [inline boa/vyper/contract.py:389]:
```suggestion
                        if isinstance(t, AddressT):
```

File: boa/vyper/contract.py [hunk 3]
```diff
@@ -633,13 +633,13 @@ def marshal_to_python(self, computation, vyper_typ):
...
         # unwrap the tuple if needed
-        if not isinstance(vyper_typ, TupleType):
+        if not isinstance(vyper_typ, TupleT):
             (ret,) = ret
 
         # eth_abi does not checksum addresses. patch this in a limited
         # way here, but for complex return types, we need an upstream
         # fix.
-        if is_base_type(vyper_typ, "address"):
+        if is_type_t(vyper_typ, AddressT):
```
Comment (charles-cooper) [inline boa/vyper/contract.py:642]:
```suggestion
        if isinstance(vyper_typ, AddressT):
```

File: boa/vyper/decoder_utils.py [hunk 4]
```diff
@@ -44,32 +46,32 @@ def __getitem__(self, subscript):
 
 
 def decode_vyper_object(mem, typ):
-    if is_bytes_m_type(typ):
+    if is_type_t(typ, BytesM_T):
```
Comment (charles-cooper) [inline boa/vyper/decoder_utils.py:49]: these all need to be changed. you can see similar case/check patterns in `vyper/codegen/core.py` and `vyper/builtins/_convert.py`

File: pyproject.toml [hunk 5]
```diff
@@ -15,7 +15,7 @@ classifiers = ["Topic :: Software Development"]
 
 # Requirements
 dependencies = [
-    "vyper >= 0.3.7",
+    "vyper @ git+https://github.com/vyperlang/vyper.git@master",
```
Comment (charles-cooper) [inline pyproject.toml:18]: this should always be pinned to a specific commit, if upstream breaks something here, boa will not be broken.

**Review by charles-cooper — APPROVED** [review]
thanks! saved me a bunch of work!

**Comment by charles-cooper** [comment]
looks like there's a couple merge conflicts though - could you please fix?

**Comment by tserg** [comment]
> looks like there's a couple merge conflicts though - could you please fix?

I have fixed the merge conflict. Oddly, CI is now failing because of missing packages. I am still figuring out why.

File: .github/workflows/main.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ jobs:
           python-version: 3.10.4
 
       - name: Install Requirements
-        run: >-
+        run: |
```
Comment (tserg) [inline .github/workflows/main.yaml:21]: Interesting, not sure why CI did not fail earlier if this was an issue.

Comment (charles-cooper) [reply .github/workflows/main.yaml:21]: oh man. nice catch

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**